sdk/storage/Azure.Storage.Common/tests/Shared/StorageTestBase.cs

```diff
@@ -155,7 +155,19 @@ private TenantConfiguration GetTestConfig(string name, Func<TenantConfiguration>
                 case RecordedTestMode.Playback:
                     if (!_playbackConfigCache.TryGetValue(name, out config))
                     {
-                        text = Recording.GetVariable(name, null);
+                        try
+                        {
+                            text = Recording.GetVariable(name, null);
+                        }
+                        catch
+                        {
+                            // If we can't find the variable, treat it as a mismatch recording.
+                            throw new TestRecordingMismatchException("no recordings found");
```

Comment:
what exception are we after here?
is `Recording` null ? if that case we should check it rather than catch.
otherwise should we move this into `Recording.GetVariable`?

Reply:
In this case, Recording is not null, but it does not contain the variable, which throws. Because this base class implements its own logic around RecordedTestMode, it doesn't allow [RecordedTest] to work properly unless the exception thrown is `TestRecordingMismatchException` (which is what this scenario is, effectively).

Reply:
In this case, Recording is not null, but it does not contain the variable, which throws. Because this base class implements its own logic around RecordedTestMode, it doesn't allow [RecordedTest] to work properly unless the exception thrown is `TestRecordingMismatchException` (which is what this scenario is, effectively).

Reply:
What exception does Recording.GetVariable throw in that case?

Reply:
I don't recall now, but it's not one of ours. But I believe if we end up here for any reason, it should be safe to treat as a recording mismatch.